services/payment_service.py

```
@@ -45,20 +96,42 @@ async def create_transaction(user_id: str, bundleId:str,tx_ref:Optional[str] = N
+            transaction = TransactionIn(
+                userId=user_id,
+                paymentId=tx_ref,
+                TransactionType=tx_type,
+                numberOfStars=0,
+                amount=payment.amount,
+            )
+            await create_transaction_history(transaction)
+        return await _update_subscription(user_id=user_id, duration_days=payment.durationDays)
 
 
 async def pay_for_chapter(user_id: str,bundle_id:str,chapter_id: str) -> Optional[UserOut]:
```

Ruff (0.14.10)

Unused function argument: `bundle_id`

(ARG001)